backend/db_interface.py

```diff
@@ -0,0 +1,52 @@
+from pymongo import MongoClient, errors
+
+VERWIKI_DB_NAME = 'verwiki'
+TREES_TABLE_NAME = "radialTrees"
+
+def get_tree_by_id(id: int) -> object:
+    """
+    Returns the tree with the specified ID from the database, or 
+    an error, which is to be handled by the caller.
+    """
+    client = validate_and_retrieve_client() # Don't handle error here; allow it to bubble up
+    output = client[VERWIKI_DB_NAME][TREES_TABLE_NAME].find_one({"id": id})
+    client.close()
+    if output == None:
+        raise KeyError("No tree found with the id " + str(id))
+    if 'data' not in output: # We should never get this; data should be validated before entering DB!
+        print("ERROR MALFORMED DATA IN DATABASE")
+        raise TypeError("Malformed data in DB for key " + str(id))
```

Comment:
This here should be KeyError no? Because a Key is not present in a Key-Value relation. Can also get rid of print if raising Exception with a print val imo?

Reply:
Also, f-strings please : )

Reply:
At this point we were already able to find an object matching the key, so it shouldn't be a KeyError. In this case, the object in the DB doesn't follow the expected format, hence the TypeError. We also should have a print here, since it is an immediate cause for concern if our DB has malformed data (Data should be verified before entering DB)

yeah, f-strings are equivalent to what I have here, but I guess I can change it; if we feel strongly about f-strings they should be in the linter

Reply:
They are equivalent, but form a much cleaner code style which abstracts out the str() call and the plus signs.

Reply:
ok, though probably not worth bothering about; do you know of any linter step that does this?